File: backend/internal/models/todo.go
package models

import (
	"time"
)

type Todo struct {
	ID          uint      `json:"id" gorm:"primaryKey"`
	Title       string    `gorm:"size:255;not null" json:"title"` // タスクのタイトル
	Description string    `gorm:"type:text" json:"description"`   // タスクの詳細（オプション）
	Completed   bool      `gorm:"default:false" json:"completed"` // 完了フラグ
	DueDate     time.Time `json:"dueDate"`                        // 締め切り日時（オプション）
	CreatedAt   int64     `json:"createdAt"`
	UpdatedAt   int64     `json:"updatedAt"`
}

File: frontend/src/app/actions/todo.ts
"use server";

import { cookies } from "next/headers";
import { redirect } from "next/navigation";

export interface FormState {
  error: string;
}

export const updateTodo = async (
  id: number,
  state: FormState,
  todo: FormData
) => {
  const newTodo = {
    title: todo.get("title"),
    description: todo.get("description"),
    completed: Boolean(todo.get("completed")),
    dueDate: todo.get("dueDate"),
  };

  console.log("updateTodo", id, newTodo);

  const response = await fetch(`${process.env.API_URL}/todos/${id}`, {
    method: "PUT",
    body: JSON.stringify(newTodo),
    headers: {
      Authorization: `Bearer ${(await cookies()).get("token")?.value}`,
    },
  });

  if (!response.ok) {
    return { ...state, error: "タスクの更新に失敗しました。" };
  }

  redirect("/");
};
